app/Http/Controllers/Table/AlertLogController.php

```diff
@@ -0,0 +1,582 @@
+            'alert_rules.name',
+        ];
+    }
+
+    protected function filterFields($request): array
+    {
+        return [
+            'alert_log.device_id' => 'device_id',
+            'alert_log.rule_id' => 'rule_id',
+            'state' => function ($query, $state) {
+                if ($state !== '-1' && $state !== '' && $state !== null) {
+                    return $query->where('alert_log.state', $state);
+                }
+                return $query;
+            },
+            'min_severity' => function ($query, $min_severity) {
+                // Map the numeric values to the actual enum values based on severity
+                if (is_numeric($min_severity) && $min_severity !== '') {
+                    $min_severity = (int) $min_severity;
+                    switch ($min_severity) {
+                        case 1:
+                            return $query->whereIn('alert_rules.severity', ['ok', 'warning', 'critical']);
+                        case 2:
+                            return $query->whereIn('alert_rules.severity', ['warning', 'critical']);
+                        case 3:
+                            return $query->where('alert_rules.severity', 'critical');
+                        case 4:
+                            return $query->where('alert_rules.severity', 'ok');
+                        case 5:
+                            return $query->where('alert_rules.severity', 'warning');
+                        default:
+                            return $query;
+                    }
+                }
+                return $query;
+            },
+        ];
+    }
+
+    protected function sortFields($request): array
+    {
+        return [
+            'time_logged' => 'alert_log.time_logged',
+            'hostname' => 'devices.hostname',
+            'alert' => 'alert_rules.name',
+            'severity' => 'alert_rules.severity',
+            'status' => 'alert_log.state',
+        ];
+    }
+
+    protected function baseQuery(Request $request): Builder
+    {
+        $query = AlertLog::query()
+            ->hasAccess($request->user())
+            ->with(['device', 'rule'])
+            ->select([
+                'alert_log.*',
+                'devices.hostname',
+                'alert_rules.severity',
+                'alert_rules.name as alert_name'
+            ])
+            ->leftJoin('devices', 'alert_log.device_id', '=', 'devices.device_id')
+            ->leftJoin('alert_rules', 'alert_log.rule_id', '=', 'alert_rules.id');
+
+
+        return $query;
+    }    /**
+     * @param  AlertLog  $alertlog
+     */
+    public function formatItem($alertlog): array
+    {
+        static $rulei = 0;
+        $rulei++;
+
+        $alert_state = $alertlog->state;
+        $fault_detail = '';
+
+        if ($alert_state == '0') {
+            // For recovered alerts, get the latest active state details
+            $last_active = AlertLog::where('device_id', $alertlog->device_id)
```

Comment:
You could probably make this a relationship so you don't have to do an n+1 query for it.  I'm not 100% sure that will work out though.